src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundBlock.WasCompilerGenerated;
-            return new CSharpLazyBlockOperation(this, boundBlock, locals, _semanticModel, syntax, type, constantValue, isImplicit);
+            var parameters = (_semanticModel as MemberSemanticModel)?.MemberSymbol.GetParameters() ?? ImmutableArray<ParameterSymbol>.Empty;
+            var nullChecksToPrepend = GenerateNullChecksForParameters(parameters, syntax);
+            return new CSharpLazyBlockOperation(this, boundBlock, locals, nullChecksToPrepend, _semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private ConditionalOperation GenerateNullCheckForParameter(ParameterSymbol parameter, SyntaxNode syntax)
+        {
+            Optional<object> constantValue = default(Optional<object>);
+            var compilation = (CSharpCompilation)_semanticModel.Compilation;
+            IOperation conditionOp;
+            if (parameter.Type.IsNullableType())
+            {
+                conditionOp = new UnaryOperation(
+                    UnaryOperatorKind.Not,
+                    new InvocationOperation(
+                        targetMethod: ((IMethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue)).Construct(parameter.Type),
```

Comment:
We need to do the same test for whether this method is present in the compilation that we do in the local rewriter version. Please add a test for this scenario as well.

Reply:
Is this similar to the MissingMember tests, or which are your referring to?